packages/design-system-react-native/src/index.ts

```diff
@@ -1,11 +1,27 @@
 import { withThemeProvider } from '@metamask/design-system-twrnc-preset';
 
-import TextComponent from './components/Text';
+import ButtonComponent from './components/Button';
+export const Button = withThemeProvider(ButtonComponent);
+export { ButtonProps, ButtonSize, ButtonVariant } from './components/Button';
+
+import ButtonIconComponent from './components/ButtonIcon';
+export const ButtonIcon = withThemeProvider(ButtonIconComponent);
+export { ButtonIconProps, ButtonIconSize } from './components/ButtonIcon';
+
+import IconComponent from './components/Icon';
+export const Icon = withThemeProvider(IconComponent);
+export { IconColor, IconName, IconProps, IconSize } from './components/Icon';
 
+import TextComponent from './components/Text';
 export const Text = withThemeProvider(TextComponent);
 export {
-  TextVariant,
-  TextColor,
   FontWeight,
   FontStyle,
+  TextColor,
+  TextProps,
+  TextVariant,
 } from './components/Text';
+
+import TextButtonComponent from './components/TextButton';
+export const TextButton = withThemeProvider(TextButtonComponent);
+export { TextButtonProps } from './components/TextButton';
```

Comment:
@georgewrmarshall I was debating whether to export non-official components (AvatarBase, ButtonBase, TextOrChildren, ButtonPrimary, ButtonSecondary, ButtonTertiary, ButtonAnimated). Let me know your thoughts on these

Reply:
Good question! I think base components are worth exporting, but any variants with a unification component probably aren't. What do you think?